### PR TITLE
[pipelineX](refactor) Rebuild relationship between dep and operator

### PR DESCRIPTION
**Pull request by Gabriel39**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by Gabriel39** [comment]
run buildall

**Comment by Gabriel39** [comment]
run buildall

**Comment by Gabriel39** [comment]
run buildall

File: be/src/pipeline/exec/scan_operator.h [hunk 1]
```diff
@@ -166,6 +167,8 @@ class ScanLocalState : public ScanLocalStateBase {
 
     RuntimeFilterDependency* filterdependency() override { return _filter_dependency.get(); };
```
Comment (yiguolei) [inline be/src/pipeline/exec/scan_operator.h:168]: 这个filter dependency 还有必要有单独的函数吗？ 直接整合到 dependencies 这个函数里一起返回是否就可以了

File: be/src/pipeline/pipeline_x/dependency.h [hunk 2]
```diff
@@ -118,21 +120,21 @@ class Dependency : public std::enable_shared_from_this<Dependency> {
     void set_ready();
     void set_ready_to_read() {
         DCHECK(_is_write_dependency) << debug_string();
-        DCHECK(_shared_state->source_dep != nullptr) << debug_string();
-        _shared_state->source_dep->set_ready();
+        DCHECK(_shared_state->source_deps.size() == 1) << debug_string();
+        _shared_state->source_deps.front()->set_ready();
```
Comment (yiguolei) [inline be/src/pipeline/pipeline_x/dependency.h:124]: 为啥这里是front？
另外，为啥shared state 的source deps 是一个vector，而不是就一个？而会有多个

**Comment by Gabriel39** [comment]
run buildall

**Review by yiguolei — APPROVED** [review]

**Review by Mryange — APPROVED** [review]

**Pull request merged**